indico/MaKaC/webinterface/pages/category.py

```diff
@@ -278,62 +269,33 @@ def __init__( self, aw, session, date, details="conference" ):
-        return conveners
-
-    def _getLocation( self ):
-        loc = ""
-        if self._session.getOwnLocation() != None:
-            loc = self._session.getLocation().getName()
-        room = ""
-        if self._session.getOwnRoom() != None:
-            if loc != "":
-                loc = "%s: "%loc
-            url = linking.RoomLinker().getURL( self._session.getRoom(), \
-                                                   self._session.getLocation() )
-            room = self._session.getRoom().getName()
-            if url != "":
-                room = """<a href="%s" style="font-size: 0.9em">%s</a>"""%(url, room)
-        if loc != "" or room != "":
-            return "(%s %s)"%(loc, room)
-        else:
-            return ""
+    def _getConvenerText(self):
+        conveners = [link.person.full_name for block in self._session.blocks for link in block.person_links]
+        return u"({})".format(u"; ".join(conveners))
 
     def _getBreakItem( self, entry ):
         wc = WOverviewBreak( self._aw, entry )
         return wc.getHTML( {} )
 
-    def _getDetails( self ):
+    def _getDetails(self):
         if self._details != "contribution":
             return ""
         res = []
-        tz = DisplayTZ(self._aw).getDisplayTZ()
-        for entry in self._session.getSchedule().getEntriesOnDay( self._date ):
+        entries = [block.timetable_entries.filter_by(start_dt=self._date) for block in self._session.blocks]
+        for entry in entries:
             if self._details == "contribution":
-                if isinstance( entry, schedule.LinkedTimeSchEntry) and \
-                        isinstance( entry.getOwner(), conference.Contribution):
-                    wc = WOverviewContribution( self._aw, \
-                                                entry.getOwner(), \
-                                                self._date, \
-                                                self._details )
-                    res.append( wc.getHTML( ) )
-                elif isinstance( entry, schedule.BreakTimeSchEntry):
-                    res.append( self._getBreakItem( entry ) )
-        return "".join( res )
+                if entry.type == TimetableEntryType.CONTRIBUTION:
+                    wc = WOverviewContribution(self._aw, entry.getOwner(), self._date, self._details)
+                    res.append(wc.getHTML())
+                elif entry.type == TimetableEntryType.BREAK:
+                    res.append(self._getBreakItem(entry))
+        return u"".join(res)
 
     def getVars( self ):
         vars = wcomponents.WTemplated.getVars( self )
         tz = DisplayTZ(self._aw).getDisplayTZ()
-        vars["startTime"] = self._session.getStartDate().astimezone(timezone(tz)).strftime("%H:%M")
-        conf=self._session.getConference()
+        vars["startTime"] = self._slot.timetable_entry.start_dt.astimezone(timezone(tz)).strftime("%H:%M")
+        conf=self._session.event_new
```

Comment:
Also, shouldn't we pass `event_new.as_legacy`? I see that `getFactory` calls `getId` on it.